File: sdks/python/paycrest_sdk/webhooks.py
import hashlib
import hmac


def verify_webhook_signature(raw_body: str, signature: str, secret: str) -> bool:
    digest = hmac.new(secret.encode("utf-8"), raw_body.encode("utf-8"), hashlib.sha256).hexdigest()
    return hmac.compare_digest(digest, signature)

File: sdks/go/sdk/webhooks.go
package sdk

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
)

func VerifyWebhookSignature(rawBody, signature, secret string) bool {
	mac := hmac.New(sha256.New, []byte(secret))
	mac.Write([]byte(rawBody))
	expected := hex.EncodeToString(mac.Sum(nil))
	return hmac.Equal([]byte(expected), []byte(signature))
}
